src/Uno.Extensions.Toolkit.UI/ThemeService.cs

```diff
@@ -21,35 +22,77 @@ internal class ThemeService : IThemeService
 		_dispatcher = dispatcher;
 		_logger = logger;
 
-		_ = _dispatcher.ExecuteAsync(async ct =>
+		Console.WriteLine($"Thread access: {_dispatcher.HasThreadAccess}");
+		if (!_dispatcher.HasThreadAccess && PlatformHelper.IsThreadingEnabled)
 		{
-			_rootAccessorElement = window.Content;
+			// Need to dispatch in order to access window.Content on UI thread
+			_ = _dispatcher.ExecuteAsync(InitWindow);
+		}
+		else
+		{
+			_ = InitWindow(CancellationToken.None);
+		}
+
+		async ValueTask InitWindow(CancellationToken ct)
+		{
+			RootElement = window.Content;
 			await InitializeAsync();
-		});
+		}
 	}
 
 	internal ThemeService(
 		UIElement rootAccessorElement,
 		IDispatcher dispatcher,
 		ILogger? logger = default)
 	{
-		_rootAccessorElement = rootAccessorElement;
+		RootElement = rootAccessorElement;
 		_dispatcher = dispatcher;
 		_logger = logger;
 
 		_ = InitializeAsync();
 	}
 
 	/// <inheritdoc/>
-	public bool IsDark => _rootAccessorElement?.XamlRoot is { } xamlRoot ? SystemThemeHelper.IsRootInDarkMode(xamlRoot) : false;
+	public bool IsDark => (RootElement as FrameworkElement)?.ActualTheme == ElementTheme.Dark;
 
 	/// <inheritdoc/>
 	public AppTheme Theme => GetSavedTheme();
 
 
+	public UIElement? RootElement
```

Comment:
Is there a reason we need this to be public? It's not part of the public API for IThemeService? Can this be private, protected or internal instead?